NewForce-Cohort-6/tabloidfullstack-cranberry-sauce
# Jg edit tag 16

# Description
[Ticket number 16](https://trello.com/c/wub4mWj8/16-16-edit-a-tag) added the  functionality for editing a tag.



# Testing Instructions

1.  `git add` and `git commit` whatever you're working on on your branch
2.  `git checkout main`
3. ` git fetch --a`
4.  `git checkout jg-EditTag-16` to switch to my branch.
5. If you don't see the most recent version of the files, try` git pull origin  jg-EditTag-16`.
6. If you haven't created the Tabloid FullStack database, please do so.
7.  Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/01_Tabloid_Create_DB.sql)
8.  Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/02_Tabloid_Seed_Data.sql)
9. Execute Tabloid from inside of Visual Studio by clicking on the solid green arrow. This will launch Swagger and run the back end of the application to provide the API that will be utilized to view the front end. 
10. From your terminal run npm start from the `client` directory within `tabloidfullstack-cranberry-sauce`
11.  Step 10 will bring up the front end of the application in a browser window titled React App. In the url of that browser page, please Login using an email that you know is there and works for you (example: foo@bar.com from seed data)

12. Navigate to the Tag Management page using the nav bar.  You should see a list of Tag names printed on the that include the following from the seed data unless you have altered the list in a previous test: 
-C#
-Funny
-JavaScript
-Scary
13. Next to each tag name is a delete and an EDIT button. Select a tag that you want to edit and click the edit button next to it. 
14. An editing field should appear next to the tag you are editing. Type the new name for that tag into the field and click SAVE. 
15. You should immediately see your change in the Tag List.  
16. Select another tag that you don’t want to edit and click the edit button next to it. 
17. When the edit form appears, click cancel.
18. You should see the edit form disappear with all the tags on the page remaining the same

Confirmation of steps 15 and 18 indicate the feature functions as intended. 



# Checklist:
